app/code/Magento/Catalog/Model/ResourceModel/Category/Tree.php

```diff
@@ -179,7 +180,8 @@ public function addCollectionData(
         $sorted = false,
         $exclude = [],
         $toLoad = true,
-        $onlyActive = false
+        $onlyActive = false,
+        $onlyIncludeInMenu = false
```

Comment:
Changing an @api public method is BIC, please have a look at https://devdocs.magento.com/contributor-guide/backward-compatible-development/

Reply:
hi @gabrieldagama  i have added code as BIC . Please review  this

Reply:
@sodhisarab please, extract logic to the separate class